binder/Utils/XamarinAndroid.cs

```diff
@@ -141,10 +141,8 @@ public static string FindAssembly(string assemblyName)
                     return GetFullPath(Combine(GetDirectoryName(javaBin), "../.."));
             }
 
-            if (string.IsNullOrEmpty(home))
-                throw new Exception("Cannot find Java SDK: JAVA_HOME environment variable is not set.");
-
-            return string.Empty;
+            //Last resort use AndroidSdk, which requires the Android SDK but has logic for finding Java on all platforms
+            return AndroidSdk.JavaSdkPath;
```

Comment:
Can this ever return null? Or will it throw itself if it can't find things?

Reply:
`AndroidSdk` throws with a reasonable error message, which is why have to use it last. It is really the most reliable, since it is actively used in Xamarin.Android.

Can't use it by default though--in the past people wanted to use the Java generator (for desktop), but didn't have the Android SDK.